core/domain/exp_domain.py

```diff
@@ -4184,6 +4195,19 @@ def create_mapping_for_new_version(
         state_id_map = StateIdMapping(
             new_exploration.id, new_exploration.version, new_state_names_to_ids,
             largest_state_id_used)
+
+        # Do one final check that all state names in new exploration are present
+        # in generated state names to id mapping.
+        if not set(new_state_names_to_ids.keys()) == set(
```

Comment:
Maybe ``if set(...) != set(...):`` instead?

Reply:
Modified it